source/deployment/ha.rst

```diff
@@ -0,0 +1,55 @@
+
+
+Deploying a multi-database configuration 
+^^^^
+
+To configure a multi-database Mattermost server edit the `DataSource` setting `config.json` with a list of comma-separated database connection strings, with the first entry being the connection to the master database.
+
+All connection strings must be compatible with the database type set in `DriverName`, which would be either `postgres` or `mysql`. 
+
+Database write requests will be sent to the master database and read requests will be distributed among the other databases in the list. 
+Loading a multi-database configuration onto an active server
+^^^^
+
+After a multi-database configuration has been defined in `config.json` the following procedure can be used to apply the settings without shutting down the Mattermost server: 
+
+1. Go to **System Console** > **Configuration** and press *Reload Configuration from Disk** to reload configuration settings for the Mattermost server from `config.json`. 
+2. Go to **System Console** > **Database** and press **Recycle Database Connections** to takedown existing database connections and set up new connections in the multi-database configuration. 
+
+While connection settings are changing there will be a brief moment when writes to the master database will be unsuccessful. End users attempting to send messages while the switch is happening will have an experience similar to losing connection to the Mattermost server.
+
+Automatic read replica failover 
+^^^^
+
+Should a read replica fail from exceed disk space, having a hardware failure, a network disconnection, or any other reason, the Mattermost server will automatically re-distribute read requests to the other read replica and master database servers. 
+
+Manual failover for master database  
+^^^^
+
+If the need arises to switch from the current master database--for example, if it is running out of disk space, or requires maintenance updates, or for other reasons--the Mattermost server can switch to using one of its read replicas as a master database by updating `DataSource` in `config.json` the following procedure can be used to apply the settings without shutting down the Mattermost server: 
+
+1. Go to **System Console** > **Configuration** and press *Reload Configuration from Disk** to reload configuration settings for the Mattermost server from `config.json`. 
+2. Go to **System Console** > **Database** and press **Recycle Database Connections** to takedown existing database connections and set up new connections in the multi-database configuration. 
+
+While connection settings are changing there will be a brief moment when writes to the master database will be unsuccessful. End users attempting to send messages while the switch is happening will have an experience similar to losing connection to the Mattermost server.
```

Comment:
"there will be a brief moment" -> "there may be a brief moment"  It does wait for all the existing connections to finish and starts serving new requests with the new connections.